R/MixedModelsCommon.R

```diff
@@ -792,1293 +793,4580 @@
+      options$plotsMappingColor,
+      options$plotsMappingShape,
+      options$plotsMappingLineType,
+      options$plotsMappingFill
+    )]
+  if (length(mapping) == 0)
+    mapping <- ""
+
+  # specify data_arg
+  if (options$plotsGeom == "geom_jitter") {
+    data_arg <- list(
+      position =
+        ggplot2::position_jitterdodge(
+          jitter.width  = options$plotJitterWidth,
+          jitter.height = options$plotJitterHeight,
+          dodge.width   = options$plotDodge
+        )
+    )
+  } else if (options$plotsGeom == "geom_violin") {
+    data_arg <- list(width = options$plotGeomWidth)
+  } else if (options$plotsGeom == "geom_boxplot") {
+    data_arg <- list(width = options$plotGeomWidth)
+  } else if (options$plotsGeom == "geom_count") {
+    data_arg <- list()
+  } else if (options$plotsGeom == "geom_beeswarm") {
+    data_arg <- list(dodge.width = options$plotDodge)
+  } else if (options$plotsGeom == "geom_boxjitter") {
+    data_arg <- list(
+      width             = options$plotGeomWidth,
+      jitter.width      = options$plotJitterWidth,
+      jitter.height     = options$plotJitterHeight,
+      outlier.intersect = TRUE
+    )
+  }
+  if (options$plotsBackgroundColor != "none" && options$plotsGeom != "geom_jitter" && "color" %in% mapping)
+    data_arg$color <- options$plotsBackgroundColor
+
+  # fixing afex issues with bootstrap and LRT type II SS - hopefully removeable in the future
+  if (type %in% c("LMM", "GLMM"))
+    if (options$method %in% c("LRT", "PB") && options$type == 2)
+      model <- model$full_model[[length(model$full_model)]]
+
+  .setSeedJASP(options)
+  p <- tryCatch(
+    afex::afex_plot(
+      model,
+      dv          = .v(options$dependentVariable),
+      x           = .v(unlist(options$plotsX)),
+      trace       = if (length(options$plotsTrace) != 0) .v(unlist(options$plotsTrace)),
+      panel       = if (length(options$plotsPanel) != 0) .v(unlist(options$plotsPanel)),
+      id          = .v(options$plotsAgregatedOver),
+      data_geom   = getFromNamespace(options$plotsGeom, geom_package),
+      mapping     = mapping,
+      error       = options$plotsCImethod,
+      error_level = options$plotsCIwidth,
+      data_alpha  = options$plotAlpha,
+      data_arg    = if (length(data_arg) != 0) data_arg,
+      error_arg   = list(
+        width = 0,
+        size  = .5 * options$plotRelativeSize
+      ),
+      point_arg   = list(size = 1.5 * options$plotRelativeSize),
+      line_arg    = list(size = .5 * options$plotRelativeSize),
+      legend_title = paste(.unv(unlist(options$plotsTrace)), collapse = "\n"),
+      dodge       = options$plotDodge
+    ),
+    error = function(e)
+      e
+  )
+
+  if (any(class(p) %in% c("simpleError", "error"))) {
+    plots$setError(p$message)
+    return()
+  }
+
+  if (options$plotsGeom == "geom_violin" && (length(options$plotsAgregatedOver) == 1 && length(unique(dataset[, .v(options$plotsAgregatedOver)])) < 3)) {
```

Comment:
```suggestion
  if (options$plotsGeom == "geom_violin" && (length(options$plotsAgregatedOver) == 1 && length(unique(dataset[, options$plotsAgregatedOver)]) < 3)) {
```
`.v` is no longer needed.

Reply:
removed